pkg/schedule/splitter/region_splitter.go

```diff
@@ -147,6 +147,9 @@ func (r *RegionSplitter) groupKeysByRegion(keys [][]byte) map[uint64]*regionGrou
 		if !r.checkRegionValid(region) {
 			continue
 		}
+		if bytes.Equal(region.GetStartKey(), key) {
```

Comment:
Do we need to check end key?

Reply:
No, we don't need it.

Reply:
If key is strart key of a region, does TiKV report error?